src/assets/scripts/airport/Arrival/ArrivalBase.js

```diff
@@ -61,150 +187,208 @@ export default class ArrivalBase {
             this.heading = degreesToRadians(options.heading);
         }
 
+        // TODO: is altitude ever not a number?
         if (typeof this.altitude === 'number') {
             this.altitude = [this.altitude, this.altitude];
         }
 
         if (options.route) {
-            this.route = options.route;
+            this.activeRouteModel = new RouteModel(options.route);
         } else if (options.fixes) {
-            for (let i = 0; i < options.fixes.length; i++) {
-                this.fixes.push({ fix: options.fixes[i] });
-            }
+            // TODO: this may not be needed at all. we could just use `_get()` instead.
```

Comment:
Added #56 to capture this refactor.  

Wether this goes away completely or is just simplified has yet to be determined, but at least there is an issue for it.